### ISSUE TITLE
Web apps fail to pull new images from ACR

### ISSUE THREAD

**Issue by yuvalyaron** [microsoft/AzureTRE#2544]
**Describe the bug**

When updating web apps in TRE , images cannot be pulled from ACR.
However, when the app is created (by terraform), the image is pulled successfully. 
When updating the tag and restarting the web app, it will fail to retrieve the image and will not rollback to the old version which will cause the API to return  HTTP 502. 

> 2022-09-04T11:40:57.260Z INFO - Attempting to pull image acryuvalymgmt.azurecr.io/microsoft/azuretre/api:0.4.25 from VNET.
2022-09-04T11:40:58.892Z ERROR - Image pull for acryuvalymgmt.azurecr.io/microsoft/azuretre/api:0.4.25 failed. UnexpectedFaliure
2022-09-04T11:40:58.903Z ERROR - Pulling docker image acryuvalymgmt.azurecr.io/microsoft/azuretre/api:0.4.25 over VNET failed.
2022-09-04T11:40:58.904Z WARN - Image pull failed. Defaulting to local copy if present.
2022-09-04T11:40:58.907Z ERROR - Image pull failed: Verify docker image configuration and credentials (if using private repository)

**Steps to reproduce**
1. make sure you have at least 2 versions (tags) of the `api` image in ACR
2. go to the Deployment Center of the `api` web app and change the `Tag` to another version that you have in ACR
3. restart the web app - which will make it pull the image
4. go to the URL of the API, or check the logs in the Deployment Center

**Comment by yuvalyaron** [comment]
I solved it on my environment by adding a private endpoint for ACR in the SharedSubnet

**Comment by yuvalyaron** [comment]
Was solved on my environment by adding a private endpoint for ACR in the SharedSubnet

**Comment by lenisha** [comment]
Same issue in our deployment  @yuvalyaron  did you find the fix or workaround?

**Comment by yuvalyaron** [comment]
@lenisha I added a private endpoint for the ACR in the shared subnet.
At first we thought that it solved it, but I had encountered it once again (even with the private endpoint) and had to delete the api web app and deploy again to resolve.
So I'm not sure that it's the solution, but seems like it's a good workaround

**Comment by lenisha** [comment]
yikes, so delete web-app and make deploy-core ? @yuvalyaron

**Comment by yuvalyaron** [comment]
@lenisha yes, but only as a last resort, I suggest adding that private endpoint first, it might work without removing the web app.
(note that you have to change the ACR plan to premium first, the UI can guide you)

**Comment by marrobi** [comment]
Not seen this for a long time so closing the issue. @yuvalyaron feel free to reopen!
